xbmc/windowing/Resolution.cpp

```diff
@@ -140,20 +139,28 @@ void CResolutionUtils::FindResolutionFromWhitelist(float fps, int width, int hei
         (info.dwFlags & D3DPRESENTFLAG_MODEMASK) == (curr.dwFlags & D3DPRESENTFLAG_MODEMASK) &&
         MathUtils::FloatEquals(info.fRefreshRate, fps, 0.01f))
     {
+      float fit_ratio = (height == info.iScreenHeight)
+                             ? static_cast<float>(width) / static_cast<float>(info.iScreenWidth)
+                             : static_cast<float>(height) / static_cast<float>(info.iScreenHeight);
+      if (fit_ratio > 1.0f)
+        fit_ratio = 1.0f / fit_ratio;
+
```

Comment:
Keep in mind, that many movies are not 1920x1080 - they are only 1920x800 or 1920x816 as users rip it that way to save the "black bars".

So in many cases height won't be iScreenHeight at all. The aspect ratio should be extracted from the movie itself.

Reply:
@fritsch, it's not about **aspect** ratio. It is used percentage of the output resolution, i.e. percentage video use in output (or 100% - percent of "black bars").